lesson_16/objects/objects_app/src/test/java/com/codedifferently/lesson16/lunch/LunchTest.java

```diff
@@ -0,0 +1,132 @@
+package com.codedifferently.lesson16.lunch;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class LunchTest {
+
+  private Lunch lunch;
+  private final PrintStream originalOut = System.out; // Store the original System.out
+
+  @BeforeEach
+  public void setUp() throws InvalidCalorieException {
+    lunch = new Lunch("Grilled Chicken", "Caesar Salad", 450, Lunch.LunchType.NON_VEGETARIAN);
+  }
+
+  @Test
+  public void testAddDrink() {
+    lunch.addDrink("Iced Tea");
+    lunch.addDrink("Lemonade");
+
+    // Verify that drinks were added correctly
+    ArrayList<String> expectedDrinks = new ArrayList<>();
+    expectedDrinks.add("Iced Tea");
+    expectedDrinks.add("Lemonade");
+
+    // Use reflection or a method to get the drinks if you have one
+    assertEquals(expectedDrinks.size(), lunch.getDrinks().size());
+    assertTrue(lunch.getDrinks().containsAll(expectedDrinks));
+  }
+
+  @Test
+  public void testGetHealthMessageHealthy() throws InvalidCalorieException {
+    lunch = new Lunch("Salad", "Fruit", 300, Lunch.LunchType.VEGAN);
+    assertEquals("This lunch is healthy!", lunch.getHealthMessage());
+  }
+
+  @Test
+  public void testGetHealthMessageUnhealthy() throws InvalidCalorieException {
+    lunch = new Lunch("Cheeseburger", "Fries", 800, Lunch.LunchType.NON_VEGETARIAN);
+    assertEquals("This lunch is high in calories.", lunch.getHealthMessage());
+  }
+
+  @Test
+  public void testDisplayDrinks() {
```

Comment:
Very clever approach to testing console output with ByteArrayOutputStream. I haven’t used it much myself, but I like how you captured the text.